### DISCUSSION TITLE
Creating infinite loop

### DISCUSSION THREAD

**Discussion by angstsmurf** [sbooth/SFBAudioEngine#202] [Q&A] [answered]
Sorry if I'm missing something obvious, but what is the preferred way to loop a sound indefinitely?

**Comment by sbooth** [comment, marked as answer]
There is no way to schedule a truly infinite repeat but you can get close using [SFBLoopableRegionDecoder](https://github.com/sbooth/SFBAudioEngine/blob/master/Decoders/SFBLoopableRegionDecoder.h). It wraps an underlying decoder (which must be seekable) and has various convenience constructors for setting up the region based on starting frame position, frame length, and repeat count. The simplest case for looping an entire decoder is something like:

```swift
let decoder = try AudioDecoder(url: url)
let loopedDecoder = try LoopableRegionDecoder(decoder: decoder, framePosition: 0, frameLength: decoder.length, repeatCount: Int.max)
// Use `loopedDecoder`
```

**Reply by angstsmurf** [reply]
Thanks!
